logstash-core/src/test/java/org/logstash/common/io/DeadLetterQueueWriterTest.java

```diff
@@ -345,6 +347,90 @@ public void testRemoveSegmentsOrder() throws IOException {
         }
     }
 
+    @Test
+    public void testUpdateOldestSegmentReference() throws IOException {
+        try (DeadLetterQueueWriter sut = DeadLetterQueueWriter
+                .newBuilderWithoutFlusher(dir, 10 * MB, 20 * MB)
+                .build()) {
+
+            final byte[] eventBytes = new DLQEntry(new Event(), "", "", "").serialize();
+
+            try(RecordIOWriter writer = new RecordIOWriter(dir.resolve("9.log"))){
+                writer.writeEvent(eventBytes);
+            }
+
+            try(RecordIOWriter writer = new RecordIOWriter(dir.resolve("10.log"))){
+                writer.writeEvent(eventBytes);
+            }
+
+            try(RecordIOWriter writer = new RecordIOWriter(dir.resolve("8.log"))){
+                writer.writeEvent(eventBytes);
+            }
+
+            // Exercise
+            sut.updateOldestSegmentReference();
+
+            // Verify
+            final Optional<Path> oldestSegmentPath = sut.getOldestSegmentPath();
+            assertTrue(oldestSegmentPath.isPresent());
+            assertEquals("10.log", oldestSegmentPath.get().getFileName().toString());
+        }
```

Comment:
Until this is fixed, I suggest writing a test that is has no surprises:

```suggestion
            try(RecordIOWriter writer = new RecordIOWriter(dir.resolve("1.log"))){
                writer.writeEvent(eventBytes);
            }

            try(RecordIOWriter writer = new RecordIOWriter(dir.resolve("2.log"))){
                writer.writeEvent(eventBytes);
            }

            try(RecordIOWriter writer = new RecordIOWriter(dir.resolve("3.log"))){
                writer.writeEvent(eventBytes);
            }

            // Exercise
            sut.updateOldestSegmentReference();

            // Verify
            final Optional<Path> oldestSegmentPath = sut.getOldestSegmentPath();
            assertTrue(oldestSegmentPath.isPresent());
            assertEquals("1.log", oldestSegmentPath.get().getFileName().toString());
        }
```